pkg/shell/authorized-keys.js

```diff
@@ -0,0 +1,228 @@
+define([
+    "jquery",
+    "base1/cockpit"
+], function($, cockpit) {
+    var module = { };
+    var _ = cockpit.gettext;
+
+    function AuthorizedKeys (user_name, home_dir) {
+        var self = this;
+        var dir = home_dir + "/.ssh";
+        var filename = dir + "/authorized_keys";
+        var file = null;
+        var watch = null;
+        var last_tag = null;
+
+        self.keys = [];
+        self.state = "loading";
+
+        function process_failure (ex) {
+            self.keys = [];
+            if (ex.problem == "access-denied") {
+                self.state = ex.problem;
+            } else if (ex.problem == "not-found") {
+                self.state = "ready";
+            } else {
+                self.state = "failed";
+                console.warn("Error proccessing authentication keys: "+ ex);
+            }
+            $(self).triggerHandler("changed");
+        }
+
+        function fix_permissions() {
+            // fix permissions, in case we created the file as root
+            cockpit.spawn(["chown", "-R", user_name+":"+user_name, dir],
+                           {'superuser': 'try'})
```

Comment:
This assumes that the user has a primary group with the same name. I would suggest avoiding that assumption. The very next line sets ```chmod 600``` which makes the group irrelevant.

In addition this has escaping issues concatenating the various values.